src/core/axum/json.rs

```diff
@@ -0,0 +1,30 @@
+use crate::APUB_JSON_CONTENT_TYPE;
+use axum::response::IntoResponse;
+use http::header;
+use serde::Serialize;
+
+/// A wrapper struct to respond with [`APUB_JSON_CONTENT_TYPE`]
+/// in axum handlers
+///
+/// ## Example:
+/// ```rust, no_run
+/// use activitypub_federation::deser::context::WithContext;
+///  async fn http_get_user() -> Result<ApubJson<WithContext<Person>>, Error> {
+///     let user = WithContext::new_default(M);
+///
+///     Ok(ApubJson(WithContext::new_default(MyUser::default())))
+/// }
+/// ```
+#[derive(Debug, Clone, Copy, Default)]
+pub struct ApubJson<Json: Serialize>(Json);
```

Comment:
The inner Json must also be public, otherwise it cannot be set from outside this crate.